services/antivirus/pkg/scanners/icap.go

```diff
@@ -89,5 +95,14 @@ func (s ICAP) Scan(in Input) (Result, error) {
 		}
 	}
 
+	if result.Infected || res.ContentResponse == nil {
+		return result, nil
+	}
+
+	// mcafee forwards the scan result as HTML in the content response;
+	// status 403 indicates that the file is infected
+	result.Infected = res.ContentResponse.StatusCode == http.StatusForbidden
+	result.Description = res.ContentResponse.Status
```

Comment:
Should we do this only if scanner is running on mcaffe? Could it break if we use another virus scanner?

Reply:
> Should we do this only if scanner is running on mcaffe? Could it break if we use another virus scanner?

i know what you mean, i thought the same... but there is no indication if the icap response is a skyHigh response or not... no header, no nothing. The only thing i could think of is to introduce a config value... but that feels maehhhh :(

the ContentResponse is always nil (clamav, mcaffe without virus), thats the reason for the early return.